File: pyg-parent/pyg-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;

import entity.PageResult;

/**
 * 品牌管理接口
 * @author admin
 *
 */
public interface BrandService {
	
	List<TbBrand> findAll();
	
	/**
	 * 品牌分页
	 * @param pageNum 当前页码
	 * @param pageSize 每页记录数
	 * @return
	 */
	PageResult findPage(int pageNum, int pageSize);
	
	/**
	 * 品牌模糊查询、分页
	 * @param brand 模糊查询条件
	 * @param pageNum 当前页码
	 * @param pageSize 每页记录数
	 * @return
	 */
	PageResult findPage(TbBrand brand, int pageNum, int pageSize);
	
	/**
	 * 添加品牌
	 * @param tbBrand
	 */
	void add(TbBrand tbBrand);
	
	/**
	 * 根据id查询品牌信息
	 * @param id
	 * @return
	 */
	TbBrand findById(long id);
	
	/**
	 * 根据id修改品牌信息
	 * @param tbBrand
	 */
	void updateBrand(TbBrand tbBrand);
	
	/**
	 * 根据id的数组删除品牌
	 * @param ids
	 */
	void delete(long[] ids);
	
	/**
	 * 返回品牌下拉列表数据
	 * @return
	 */
	List<Map> selectOptionList();
}

File: pyg-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function($http){
		
			this.findAll=function(){
				return $http.get("../brand/findAll.do");
			}
			this.findPage=function(page,size){
				return $http.get('../brand/findPage.do?page='+page+'&size='+size);
			}
			this.search=function(page,size,searchEntity){
				return $http.post('../brand/search.do?page='+page+'&size='+size,searchEntity);
			}
			this.addBrand=function(entity){
				return $http.post('../brand/addBrand.do',entity);
			}
			this.updateBrand=function(entity){
				return $http.post('../brand/updateBrand.do',entity);
			}
			this.findById=function(id){
				return $http.get('../brand/findById.do?id='+id);
			}
			this.delBrands=function(ids){
				return $http.get('../brand/deleteBrand.do?ids='+ids);
			}
			this.selectOptionList=function(){
				return $http.get('../brand/selectOptionList.do');
			}
	})